From 9a40a4d7a5cd6ce6f15a2e8fb65ee60556eff660 Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Fri, 06 Jun 2025 13:54:18 +0300
Subject: [PATCH] [SW-228856] fit: remove temp name

Context:
We encountered several code duplications related to
fit_foreach_property.
So we created a new fit services infrastructure
to solve this issue in a clean way.
The main Lego pieces are the newly created files:
fit_services.c/h

Current commit:
In this current commit, we remove any trace
(declaration, definition, invoke) of the temporary name
_fit_foreach_property.

General remark:
This infrastructure is intended to support ANY of the
fit related functions that we find today in fit_utils.c
or fit.c
Therefore, from now, please, anyone doing a change related
to the fit files should carefully double-check:
Any unified code can be (and should be) located in
fit_services.c/h
For this, please, follow the example of how it has been
done with fit_foreach_property.

Change-Id: Ia62773ae1ebee8731d664fd76bbb9d79c2a6a2b0
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/include/fit_services.h b/include/fit_services.h
index 4245c4d..fa010fe 100644
--- a/include/fit_services.h
+++ b/include/fit_services.h
@@ -12,12 +12,6 @@
 #include "libfdt.h"
 #include "fdt.h"
 
-typedef int (*_fit_property_match_cb)(const void *fit,
-		const struct fdt_property *prop, void *data);
-
-int _fit_foreach_property(const void *fit, const char *node_path,
-			_fit_property_match_cb match_cb, void *data);
-
 typedef int (*fit_property_match_cb)(const void *fit,
 		const struct fdt_property *prop, void *data);
 
diff --git a/lib/common/fit_services.c b/lib/common/fit_services.c
index dc797eb..e552c77 100644
--- a/lib/common/fit_services.c
+++ b/lib/common/fit_services.c
@@ -12,34 +12,6 @@
 #include <fit_services.h>
 #include <emb_cmn_log.h>
 
-int _fit_foreach_property(const void *fit, const char *node_path,
-			_fit_property_match_cb match_cb, void *data)
-{
-	int ret, count = 0;
-	int noffset, property;
-	const struct fdt_property *prop;
-
-	noffset = fdt_path_offset(fit, node_path);
-	if (noffset < 0) {
-		EMB_LOG_RAW_ERR("Cannot find entry ");
-		EMB_LOG_RAW(node_path);
-		EMB_LOG_RAW("\n\r");
-		return noffset;
-	}
-
-	fdt_for_each_property_offset(property, fit, noffset) {
-		prop = fdt_get_property_by_offset(fit, property, NULL);
-		if (!prop)
-			continue;
-
-		ret = match_cb(fit, prop, data);
-		if (!ret)
-			count++;
-	}
-
-	return count;
-}
-
 int fit_foreach_property(const void *fit, const char *node_path,
 			fit_property_match_cb match_cb, void *data)
 {
diff --git a/lib/common/fit_utils.c b/lib/common/fit_utils.c
index 0cd864e..fa21462 100644
--- a/lib/common/fit_utils.c
+++ b/lib/common/fit_utils.c
@@ -85,8 +85,9 @@
 {
 	int ret;
 
-	ret = _fit_foreach_property(fit, "/", (_fit_property_match_cb)fit_match_prop_and_print,
-				    HL_FIT_VERSION);
+	ret = fit_foreach_property(fit, "/", (fit_property_match_cb)fit_match_prop_and_print,
+				   HL_FIT_VERSION);
+
 	if (ret <= 0)
 		EMB_LOG_WRN("[FIT] No version properties were found.");
 }
diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 8468a7c..8b9a7c7 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -239,7 +239,7 @@
 {
 	int ret;
 
-	ret = _fit_foreach_property(fit, "/", match_prop_version, "version");
+	ret = fit_foreach_property(fit, "/", match_prop_version, "version");
 	if (ret <= 0)
 		EMB_LOG_WRN("[FIT] No version properties were found.");
 }
